pkg/common/support/collconfigretriever.go

```diff
@@ -20,6 +22,43 @@ import (
 	"github.com/pkg/errors"
 )
 
+type collectionConfigRetrieverCache struct {
+	mutex      sync.RWMutex
+	retrievers map[string]*CollectionConfigRetriever
+}
+
+var collConfigRetrieverCache = &collectionConfigRetrieverCache{
+	retrievers: make(map[string]*CollectionConfigRetriever),
+}
+
+// CollectionConfigRetrieverForChannel returns the collection config retriever for the given channel
+func CollectionConfigRetrieverForChannel(channelID string) *CollectionConfigRetriever {
+	return collConfigRetrieverCache.forChannel(channelID)
```

Comment:
I think it's ok for it to panic since this would be indicative of a bug. When a peer joins a channel then it should initialize the collection config retriever for the channel.

Reply:
sounds good.. I'm ok with this